qiskit/algorithms/evolvers/classical_methods/scipy_real_evolver.py

```diff
@@ -0,0 +1,328 @@
+
+        .. math::
+
+            |\Psi ( t + \delta t ) \rangle = \exp(-i \delta t H) |\Psi (t ) \rangle
+
+        Which can be approximated as:
+
+        .. math::
+
+            \exp(-i \delta t H) = \left(1+ i \frac{\delta t}{2} H \right) ^{-1}
+                                  \left( 1 - i \frac{\delta t}{2} H \right)
+                                  + O\left( \delta t^3 \right)
+
+        Note that this operator is unitary, and thus we won't need to renormalize the state at
+        each step. In order to find :math:`|\Psi ( t + \delta t ) \rangle` we then
+        need to solve a linear system of equations:
+
+        .. math::
+
+            \left( 1+ i \frac{\delta t}{2} H \right) |\Psi ( t + \delta t ) \rangle
+                    = \left( 1 - i \frac{\delta t}{2} H \right) |\Psi( t ) \rangle
+
+        Args:
+            evolution_problem: The definition of the evolution problem.
+
+        Returns:
+            Evolution result which includes an evolved quantum state.
+
+        Raises:
+            ValueError: If the hamiltonian is time dependent.
+
+        """
+
+        if evolution_problem.t_param is not None:
+            raise ValueError("Time dependent hamiltonians are not currently supported.")
+
+        (
+            state,
+            lhs_operator,
+            rhs_operator,
+            aux_ops,
+            aux_ops_2,
+            timesteps,
+            bicg_tol,
+        ) = self._start(evolution_problem=evolution_problem)
+
+        # Create empty arrays to store the time evolution of the aux operators.
+        operator_number = (
+            0 if evolution_problem.aux_operators is None else len(evolution_problem.aux_operators)
+        )
+        ops_ev_mean = np.empty(shape=(operator_number, timesteps + 1), dtype=float)
+
+        ops_ev_std = np.empty(shape=(operator_number, timesteps + 1), dtype=float)
+
+        # Perform the time evolution and stores the value of the operators at each timestep.
+        for ts in range(timesteps):
+            (ops_ev_mean[:, ts], ops_ev_std[:, ts],) = self._evaluate_aux_ops(
+                aux_ops,
+                aux_ops_2,
+                state,
+            )
+
+            state = self._step(state, lhs_operator, rhs_operator, bicg_tol)
+
+        (
+            ops_ev_mean[:, timesteps],
+            ops_ev_std[:, timesteps],
+        ) = self._evaluate_aux_ops(aux_ops, aux_ops_2, state)
+
+        aux_ops_history = self._create_observable_output(
+            ops_ev_mean,
+            ops_ev_std,
+            evolution_problem.aux_operators,
+        )
+        aux_ops = self._create_observable_output(
+            ops_ev_mean[:, timesteps],
+            ops_ev_std[:, timesteps],
+            evolution_problem.aux_operators,
+        )
+
+        return EvolutionResult(
+            evolved_state=StateFn(state), aux_ops_evaluated=aux_ops, observables=aux_ops_history
+        )
+
+    def _create_observable_output(
```

Comment:
This and the following method are stateless and used in both the real and imaginary evolver. Would it make sense to put that in a separate file?